subsys/task_wdt/task_wdt_auto_init.c

```diff
@@ -0,0 +1,13 @@
+/* Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/task_wdt/task_wdt.h>
+
+static int init(void)
```

Comment:
Use a more specific name, this might shadow others/macros.

```suggestion
static int task_wdt_auto_init(void)
```